storage/storage_dest.go

```diff
@@ -707,7 +753,7 @@ func (s *storageImageDestination) commitLayer(index int, info addedLayerInfo, si
 	if !ok {
 		// Try to find the layer with contents matching that blobsum.
 		layer := ""
-		layers, err2 := s.imageRef.transport.store.LayersByUncompressedDigest(diffIDOrTOCDigest)
+		layers, err2 := s.imageRef.transport.store.LayersByUncompressedDigest(layerID)
```

Comment:
`layerID`, right now, is not an uncompressed digest; this is again mixing the namespaces.

I know… we “shouldn’t” get here because we already checked `diffOutputs`, but that could have failed e.g. because some concurrent process deleted the chunked layer in the meantime.

Reply:
but we have filled `diffOutputs` before entering this function, so it should not matter if some concurrent process deleted the layer.  Not sure exactly what error we'd get if this happens but `diffOutput, ok := s.diffOutputs[info.digest]` will be true.

Reply:
Oh, you are right, we can not reach it with the wrong value.

Still, I think doing a `diffID, ok := blobDiffIDs[…]; if !ok { fail }` lookup is trivially cheap, and it would make it clear that we are using the right value.

Reply:
Also, assuming https://github.com/containers/image/pull/2218#discussion_r1424655287 , `LayersByUncompressedDigest(layerID)` will become a type error.